js/run_js_binary.bzl

```diff
@@ -106,41 +116,54 @@ def run_js_binary(
         )
         extra_srcs = [":%s" % copy_to_bin_name]
 
-    # Automatically add common and useful make variables to the environment for js_binary targets
-    # under rules_js
+    # Automatically add common and useful make variables to the environment for run_js_binary build targets
     extra_env = {
         "BAZEL_BINDIR": "$(BINDIR)",
         "BAZEL_BUILD_FILE_PATH": "$(BUILD_FILE_PATH)",
-        "BAZEL_VERSION_FILE": "$(VERSION_FILE)",
+        "BAZEL_COMPILATION_MODE": "$(COMPILATION_MODE)",
         "BAZEL_INFO_FILE": "$(INFO_FILE)",
+        "BAZEL_TARGET_CPU": "$(TARGET_CPU)",
         "BAZEL_TARGET": "$(TARGET)",
+        "BAZEL_VERSION_FILE": "$(VERSION_FILE)",
         "BAZEL_WORKSPACE": "$(WORKSPACE)",
-        "BAZEL_TARGET_CPU": "$(TARGET_CPU)",
-        "BAZEL_COMPILATION_MODE": "$(COMPILATION_MODE)",
     }
 
     # Configure working directory to `chdir` is set
-    chdir_prefix = ""
     if chdir:
         extra_env["JS_BINARY__CHDIR"] = chdir
-        chdir_prefix = "/".join([".."] * len(chdir.split("/"))) + "/"
 
     # Configure capturing stdout, stderr and/or the exit code
     extra_outs = []
     if stdout:
-        extra_env["JS_BINARY__CAPTURE_STDOUT"] = "%s$(rootpath %s)" % (chdir_prefix, stdout)
+        extra_env["JS_BINARY__STDOUT_OUTPUT_FILE"] = "$(execpath {})".format(stdout)
         extra_outs.append(stdout)
     if stderr:
-        extra_env["JS_BINARY__CAPTURE_STDERR"] = "%s$(rootpath %s)" % (chdir_prefix, stderr)
+        extra_env["JS_BINARY__STDERR_OUTPUT_FILE"] = "$(execpath {})".format(stderr)
         extra_outs.append(stderr)
     if exit_code_out:
-        extra_env["JS_BINARY__CAPTURE_EXIT_CODE"] = "%s$(rootpath %s)" % (chdir_prefix, exit_code_out)
+        extra_env["JS_BINARY__EXIT_CODE_OUTPUT_FILE"] = "$(execpath {})".format(exit_code_out)
         extra_outs.append(exit_code_out)
 
     # Configure silent on success
     if silent_on_success:
         extra_env["JS_BINARY__SILENT_ON_SUCCESS"] = "1"
 
+    # Configure log_level if specified
+    if log_level:
```

Comment:
too bad this logic gets repeated so much

Reply:
Its not likely to change in the future and its just the two spots js_binary and its wrapper macro run_js_binary so should be just limited to those. Tho, I can abstract some more to reduce the number of duplicate lines of code...

Reply:
Much cleaner
```
    # Configure log_level if specified
    if log_level:
        for env in _js_binary_envs_for_log_level(log_level):
            extra_env[env] = "1"
```
and
```
    for env in envs_for_log_level(ctx.attr.log_level):
        envs.append(_ENV_SET_IFF_NOT_SET.format(
            var = env,
            value = "1",
        ))
```